### PR TITLE
get template by classifier returns first match rather than list of all matching

### PR DESCRIPTION
**Pull request by jwgwalton**

### CODE REVIEW

File: application/controllers/template_controller.py [hunk 1]
```diff
@@ -25,14 +25,23 @@ def _get_template_by_id(template_id):
...
+            .first()
     except SQLAlchemyError:
         logger.exception('Unable to retrieve template with classifiers', classifiers=classifiers)
         raise DatabaseError(f'Unable to retrieve template with classifiers: {json.dumps(classifiers)}', status_code=500)
-    return templates
+    return template
+
+
+def get_comms_template_by_classifiers(classifiers=None):
+    template = get_template_by_classifiers(classifiers)
```
Comment (JamesGardiner) [inline application/controllers/template_controller.py:39]: I'm assuming DatabaseError is handled higher up?

File: tests/controllers/test_template_controller.py [hunk 2]
```diff
@@ -136,12 +136,12 @@ def test_create_template(self, mock_db, mock_get_template_by_id, mock_get_classi
...
         # Given there is an error connecting with the database
         mock_db.session.query = mock.MagicMock(side_effect=SQLAlchemyError)
 
         # When we try to get a template by classifier
         # Then it raises a database error
         with self.assertRaises(DatabaseError):
-            template_controller.get_templates_by_classifiers(classifiers={"GEOGRAPHY": "NI",
-                                                                          "INDUSTRY": "construction"})
+            template_controller.get_comms_template_by_classifiers(classifiers={"GEOGRAPHY": "NI",
+                                                                               "INDUSTRY": "construction"})
```
Comment (JamesGardiner) [inline tests/controllers/test_template_controller.py]:
```JSON
{"GEOGRAPHY": "NI",
 "INDUSTRY": "construction",
})
```

Keeping a `,` at the end of dicts gives a cleaner look to future pull request code comparisons.

**Review by JamesGardiner — DISMISSED** [review]

File: application/controllers/template_controller.py (hunk 1 above)
Comment (jwgwalton) [reply application/controllers/template_controller.py:39]: In the flask error handlers.

File: API.md [hunk 3]
```diff
@@ -35,31 +35,22 @@ The following endpoints are associated with the classification types that are us
 
 * `PUT /templates/cb0711c3-0ac8-41d3-ae0e-567e5ea1ef87` with a valid template object in the request json.
 
-## Get templates by classifiers
+## Get a template by classifiers
 
-* `GET /templates` will return a list of communications templates which match the classifiers, these are passed in the request json.
+* `GET /templates` will return a communications templates which matches the classifiers, these are passed in the request json.
```
Comment (benjefferies) [inline API.md]: a communications templates -> a communications template

**Review by benjefferies — APPROVED** [review]
Looks good apart from the minor typo

**Review by jcox-dev — APPROVED** [review]

**Pull request merged**